mozilla-services/socorro
# bug-1809726: update to django 4.2

This updates Socorro from Django 3.2 LTS to 4.2 LTS. This is a more involved update than Tecken was because Socorro uses more of Django's stuff for the web ui whereas Tecken is predominantly a react app.

This starts with updating Django which then required me to update django-pipeline. That unearthed some problems one of which was fixed by adjusting the bundle specification. The other required us to switch from one jsonview js lib which isn't maintained to another one which sort of is.

On top of that, Django deprecated a bunch of stuff that were were doing with the plan of removing those things in Django 5.0. So I ran `django-upgrade` and I also fixed some things by hand.

To test:

1. run `make build` which builds everything and also builds the static assets
2. run the tests
3. run the webapp, process some crash reports, click around
   1. check report view "Raw data and minidumps" and "Telemetry Environment" tabs which use the jsonview library
   2. check the admin
